GPU/Common/ShaderTranslation.cpp

```diff
@@ -307,8 +303,24 @@ bool TranslateShader(std::string *dest, ShaderLanguage destLang, TranslatedShade
 		*dest = glsl.compile();
 		return true;
 	}
+	case GLSL_300:
+	{
+		spirv_cross::CompilerGLSL glsl(std::move(spirv));
+		// The SPIR-V is now parsed, and we can perform reflection on it.
+		spirv_cross::ShaderResources resources = glsl.get_shader_resources();
+		// Set some options.
+		spirv_cross::CompilerGLSL::Options options;
+		if (gl_extensions.ver[0] >= 4) {
+			options.version = 400;
+		} else {
+			options.version = 300;
```

Comment:
I had an issue on Mac where I had to specify exactly the same version.  Maybe I was mistaken and it was only the major version, but it seemed to need 1:1.

That's what motivated this code:
https://github.com/hrydgard/ppsspp/blob/33e500a6817b7d5dd0daa9c663cde2ac38ca48b9/ext/native/gfx_es2/gpu_features.cpp#L501

I'm also pretty sure 3.1 and 3.2 needed those specific versions too (you can test by hardcoding versions in PCMain iirc.)

-[Unknown]

Reply:
Seems odd that it would be that strict about version, especially if we specify just the major version... but maybe worth doing. Should be noted that GLSL language versions are distinct from GL versions although they did synchronize them after 3.0 I think?

Reply:
Right - this path is only non-ES right?

-[Unknown]

Reply:
In practice mostly yeah - it's not used on Android. Though, I guess there's Raspberry Pi and things like that, and iOS...  Should probably tighten it up a little.